### ISSUE TITLE
transaction list DECK -c  and transaction list DECK -c -w comparison.

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#203]
The `transaction list ATTokenNewSpec2 -c -t` outputs 47 transactions on my side, while `transaction list ATTokenNewSpec2 -c -w -t` only 38.
However from the `transaction list -h` it wasn't clear to me what is the meaning of the `transaction list DECK -c` output (i.e. without the `-w` flag), when the `-o` flag is also not specified.
_
I've also run `transaction list PoBSimplest -c` and got the following output:
```
{
    'TX ID': 'a2f6a3dbad17afe36039a902dca9dc4ed01ba655964167aa535429933b69147a',
    'Burn transaction': '87c1b3138cde26cce621377d23b0fb4ec052c34cb77cc42ef11fb7b208f5526b',
    'Token amount(s)': [300.0],
    'Receiver(s)': ['mkfuTykT6anzsRjYH7Lktc2hFuTqcGpha7'],
    'Block height': 165489
}
{
    'TX ID': 'e31c7abdc9b8379c4698557a3a781d94e7ffcbd60ef4a8ea2d4f74a068709418',
    'Burn transaction': '8be48f92244dbe49fd2bee66878c93622d98f6448fd2fb4010b7c74dbe6cca04',
    'Token amount(s)': [1.0],
    'Receiver(s)': ['n3MtPoPaAREU5GEhAErBPuju83bVog2opz'],
    'Block height': 486216
}
{
    'TX ID': 'b752aa0aee690f8554d649889689f3735554a68a6311207eef67b2d7f19306a6',
    'Burn transaction': '7c8db9d1b18083b0cf440ec2b8ca0bee56afb55f5398ba185a2d80123fec4743',
    'Token amount(s)': [1.1],
    'Receiver(s)': ['n3MtPoPaAREU5GEhAErBPuju83bVog2opz'],
    'Block height': 488907
}
{
    'TX ID': 'f70f0118a5b255c042b033b50c8a0857decac77b6d46092e4ff5cfe754ae5bfc',
    'Burn transaction': '57d9d21e15cc3a624c532ccda2375fe7ae8985d6af6eb642b4d4f4d1534531e7',
    'Token amount(s)': [0.0],
    'Receiver(s)': ['n3qWzoBX24abCSq8hMMzmCb8dh5ePnhdmL'],
    'Block height': 498308
}
```
While the `transaction list PoBSimplest -c -w` reports as following:
```
{
    'TX ID': 'e31c7abdc9b8379c4698557a3a781d94e7ffcbd60ef4a8ea2d4f74a068709418',
    'Burn transaction': '8be48f92244dbe49fd2bee66878c93622d98f6448fd2fb4010b7c74dbe6cca04',
    'Token amount(s)': [1.0],
    'Receiver(s)': ['n3MtPoPaAREU5GEhAErBPuju83bVog2opz'],
    'Block height': 486216
}
{
    'TX ID': 'b752aa0aee690f8554d649889689f3735554a68a6311207eef67b2d7f19306a6',
    'Burn transaction': '7c8db9d1b18083b0cf440ec2b8ca0bee56afb55f5398ba185a2d80123fec4743',
    'Token amount(s)': [1.1],
    'Receiver(s)': ['n3MtPoPaAREU5GEhAErBPuju83bVog2opz'],
    'Block height': 488907
}
{
    'TX ID': 'f70f0118a5b255c042b033b50c8a0857decac77b6d46092e4ff5cfe754ae5bfc',
    'Burn transaction': '57d9d21e15cc3a624c532ccda2375fe7ae8985d6af6eb642b4d4f4d1534531e7',
    'Token amount(s)': [0.0],
    'Receiver(s)': ['n3qWzoBX24abCSq8hMMzmCb8dh5ePnhdmL'],
    'Block height': 498308
}
```
So the difference is the transaction `a2f6a3dbad17afe36039a902dca9dc4ed01ba655964167aa535429933b69147a`.
_
The difference between `transaction list ATTokenNewSpec2 -c` and `transaction list ATTokenNewSpec2 -c -w` are the following transactions:
- 97e82554217ff486343833c1e8c7629d459b9a831496091da46b4096d2523815
- 909863ab0e1c0f62cc0c6721bea140c2d0618d9dd7f4e7bd05ff735ce92cc1a6
- 25e8b47ba2c3af7855795ddef71bad216d1e97e6b2896915f0c6a2e8959d6f98
- 278454001e425bb605f063478f05f9f40f165c3abdb9f87304e35fe2090eeef5
- 4b73bf14988847d0760b0bad08037bb72028ee755252b9f969889b3a7b45d51d
- f4e26f3a14a7fbe7ea45bd60ad9005ebbaa36a3cecd864c7f3ef517c5fe27c86
- 3d7f4a2f74f66e797f135aa75986e36b41e60bf3e2a9f24d6f043c3488b0753e
- b5d8f64d478bc88454b378ca9f28f980bae5a9c6529c65cc5b1d3333116a58b1
- 39fab9240e6351df131b6234186ba5f0a5005ca0d86134db9bdd2ad81c892c67
_
I don't really have an assumption about why I'm getting these results.

**Comment by d5000** [comment]
I've already explained that earlier. `-c -w` outputs transactions where one of the addresses in your wallet is the sender of the burn, gateway or donation transaction. The mode `-c` without `-w` nor an address instead outputs all claim transactions from all investors/donors.

Can be closed.

**Comment by buhtignew** [comment]
I've rechecked the transactions mentioned in the OP and have discovered that the transactions `25e8b47ba2c3af7855795ddef71bad216d1e97e6b2896915f0c6a2e8959d6f98`, `f4e26f3a14a7fbe7ea45bd60ad9005ebbaa36a3cecd864c7f3ef517c5fe27c86` and `3d7f4a2f74f66e797f135aa75986e36b41e60bf3e2a9f24d6f043c3488b0753e` are displayed in my `transaction list ATTokenNewSpec2 -c` output and not in the `transaction list ATTokenNewSpec2 -c -w`.
However they seem to "touch"my wallet since the `transaction show 25e8b47ba2c3af7855795ddef71bad216d1e97e6b2896915f0c6a2e8959d6f98 -s` and the `transaction show 3d7f4a2f74f66e797f135aa75986e36b41e60bf3e2a9f24d6f043c3488b0753e -s` outputs contain the address `mtmKbmMHdcXXepscuNrwAfjsKLeZzdZdya` while the `transaction show f4e26f3a14a7fbe7ea45bd60ad9005ebbaa36a3cecd864c7f3ef517c5fe27c86 -s` output contains `n2nSbZYLV77NFHSD3P4ZdFKyp65rtTHP1b` and `mixcqdAy4aFtuBV7uRebqBTjiFWCcS4UEp`
All of the above addresses are part of my `address list -i` output and not of the `address list -p`, so they seem to be the addresses of my wallet that are not P2TH.

**Comment by d5000** [comment]
The behaviour of `-c -w` regarding which addresses belong to the wallet was indeed not identic to `-g -w` and `-b -w`, so I changed it. This means the mtmKbmMHdcXXepscuNrwAfjsKLeZzdZdya address will now be included as a sender (tried it with your wallet).

Commit is f93239d. I will soon push it to the repo as I'm currently working on the block locator issues and part of them are still completely untested so I prefer uploading all at once. 

Regarding transaction f4e26f3a14a7fbe7ea45bd60ad9005ebbaa36a3cecd864c7f3ef517c5fe27c86, I wrote already in the last post that to be shown in the `-c -w` option the _sender_ must be in your wallet. This is not the case in this transaction which was sent by my address mkfuTykT6anzsRjYH7Lktc2hFuTqcGpha7.
